packages/task/src/browser/task-configuration-manager.ts

```diff
@@ -225,8 +225,9 @@ export class TaskConfigurationManager {
 
     protected readonly toDisposeOnDelegateChange = new DisposableCollection();
     protected updateWorkspaceModel(): void {
-        const newDelegate = this.workspaceService.saved ? this.workspacePreferences : this.folderPreferences;
-        const effectiveScope = this.workspaceService.saved ? TaskScope.Workspace : this.workspaceService.tryGetRoots()[0]?.resource.toString();
+        const isFolderWorkspace = this.workspaceService.opened && !this.workspaceService.saved;
```

Comment:
The notion of `saved` vs `unsaved` workspaces wasn't clear at first to describe workspace folders versus multi-workspace files. I'm wondering if it is worth documenting better from `WorkspaceService` itself.

Reply:
👍  I've added a JSDoc explaining the conditions under which the `.saved` field is true or false.